### PR TITLE
NN-3407:

### PR DESCRIPTION
**Pull request by JamieDraperUK**
- Fix regression in returning a recurring appointment

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt [hunk 1]
```diff
@@ -78,18 +78,19 @@ class AppointmentService(
     val mainAppointmentDetails: PrisonAppointment = prisonApiService.getPrisonAppointment(appointmentId)
       ?: throw EntityNotFoundException("Appointment $appointmentId does not exist")
 
-    val offenderNo = prisonApiService.getOffenderNoFromBookingId(mainAppointmentDetails.bookingId)
+    val offenderNo = try {
+      prisonApiService.getOffenderNoFromBookingId(mainAppointmentDetails.bookingId)
+    } catch (e: Exception) {
+      null
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt:84]: We need a good reason to ignore exceptions - is it the case here that this is just extra data that shouldn't stop the rest of the data from being returned?

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt [hunk 2]
```diff
@@ -109,8 +110,14 @@ class AppointmentService(
...
     createAppointmentSpecification.repeat?.let {
+
+      val recurringAppointmentIds: Set<Long> = appointmentCreated.recurringAppointmentEventIds?.toSet() ?: emptySet()
+
+      val appointmentIds =
+        setOf(appointmentCreated.appointmentEventId).union(recurringAppointmentIds)
+
       val recurringAppointment =
-        makeMainRecurringAppointment(appointmentCreated, createAppointmentSpecification.repeat)
+        makeRecurringAppointment(appointmentIds = appointmentIds, repeat = createAppointmentSpecification.repeat)
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt:120]: That first "=" seems superfluous - does Kotlin need that?

**Review by sp-watson — APPROVED** [review]

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt (hunk 1 above)
Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt:84]: I did some testing and for some offenders I was getting a 404. I'll do some more testing and maybe remove or add some logging.

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt (hunk 2 above)
Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt:120]: Good point will update in the next PR

**Pull request merged**